### DISCUSSION TITLE
Prevent Image Scaling on High DPI Displays

### DISCUSSION THREAD

**Discussion by Razunter** [vladmandic/sdnext#1217] [Ideas]
Currently, non-100% DPI images are scaled by the browser, making them blurrier (especially in the lightbox). And I think there is no native way around this. There is a JS hack with transform suggested [here](https://stackoverflow.com/questions/48508180/prevent-image-scaling-on-highdpi-displays), but a better way would be to set scaled width\height of the image in px.

How it looks now:
![Screenshot 2023-05-30 at 17-48-54 SD Next](https://github.com/vladmandic/automatic/assets/362900/58963114-d5c1-43d1-a68f-5c477cb9253d)

How it should look:
![Screenshot 2023-05-30 at 17-50-11 SD Next](https://github.com/vladmandic/automatic/assets/362900/246d2298-dd9d-481a-bee5-819527ba3365)

**Comment by vladmandic** [comment]
i'm aware of the problem (see <https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio>), but the problem is that values other than 1 and 2 are non-standard and browser and platform dependent. and it also impacts canvas elements, not just image elements. also behavior changes depending on browser settings, for example see `chrome://flags/#enable-pixel-canvas-recording`

all-in-all, maybe we'll revisit it in the future when we have a full-fledged image viewer.
